### ISSUE TITLE
Enhancement: Carthage should be able to follow relative GitHub paths

### ISSUE THREAD

**Issue by dfed** [Carthage/Carthage#2600]
* carthage install method: [ ] .pkg, [x] homebrew, [ ] source
* `which carthage`:  `/usr/local/bin/carthage`
* `carthage version`: 0.31.0
* `xcodebuild -version`: `Xcode 10.0` `Build version 10A255`
* Are you using `--no-build`? No
* Are you using `--no-use-binaries`? No
* Are you using `--use-submodules`? No
* Are you using `--cache-builds`? No
* Are you using `--new-resolver`? No

**Cartfile**
```
github "Square/Valet" "3.2.0"
```

**Carthage Output**
```
*** Checking out Valet at "3.2.0"
A shell task (/usr/bin/env git clone --quiet ../../dfed/XCTest-watchOS.git /Users/dan_federman/Documents/source/valettest/Carthage/Checkouts/Valet/Tests/XCTest-watchOS) failed with exit code 128:
fatal: repository '../../dfed/XCTest-watchOS.git' does not exist
```

**Actual outcome**
Carthage fails to clone repository that has a submodule defined with relative path.

**Expected outcome**
Carthage should be able to translate `../../dfed/XCTest-watchOS.git` from the repository `https://github.com/square/valet.git` into `https://github.com/dfed/XCTest-watchOS.git`.

Relative GitHub paths in submodules allow the submodule to be checked out via the same mechanism as the original repo; if the top-level repo is checked out with `ssh`, the submodule will be checked out with `ssh` – similarly, if the top-level repo is checked out with `https`, the submodule will be checked out with `https`. This is useful functionality that Carthage prevents us from taking advantage of.

Note I was able to fix this issue by making the git submodule [explicitly point to the https clone](https://github.com/square/Valet/pull/156). We were turned onto this issue when a user of Valet pointed out that our [3.2.0 release failed to clone via Carthage](https://github.com/square/Valet/issues/155).

Note that once the above PR in Valet lands, Carthage will work with the Cartfile:
```
github "Square/Valet" "3.2.1"
```
Prior to the above PR merging, you can use the following Cartfile to test:
```
github "Square/Valet" "dfed--fix-carthage"
```

**Comment by dfed** [comment]
This issue is still relevant.

**Comment by blender** [comment]
Indeed! Do you want to give this a shot @dfed ? [It's very easy to get started](https://github.com/Carthage/Carthage/blob/master/CONTRIBUTING.md#get-started)

**Comment by dfed** [comment]
Solid conscription effort @blender 😉. Yeah I can take this on, but to be honest I likely won't have time until December. If you're cool with that kind of lag (which I guess is better than closing outright), let's assign this issue to me.

**Comment by blender** [comment]
Any help is appreciated!

**Comment by AlanRosenthal** [comment]
@dfed any update on this issue? I recently hit it :\

**Comment by dfed** [comment]
Shoot. Admittedly I lost track of this item, and I no longer utilize Carthage in my primary project. Let's unassign me 😬. Apologies for the churn there.

**Comment by AlanRosenthal** [comment]
cc @tmspzz

**Comment by AlanRosenthal** [comment]
happens to the best of us @dfed
